modules/core/shared/src/main/scala/lucuma/core/model/sequence/Visit.scala

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.model.sequence
+
+import cats.Eq
+import cats.syntax.all._
+import eu.timepit.refined.auto._
+import lucuma.core.util.WithUid
+import monocle.Focus
+import monocle.Lens
+import monocle.Prism
+import monocle.macros.GenPrism
+import org.typelevel.cats.time._
+
+import java.time.Duration
+import java.time.Instant
+
+sealed trait Visit {
+  def id: Visit.Id
+  def created: Instant
+  def staticConfig: StaticConfig
+  def steps: List[StepRecord]
+
+  def startTime: Option[Instant] = steps.headOption.flatMap(_.startTime)
+  def endTime: Option[Instant]   = steps.lastOption.flatMap(_.endTime)
```

Comment:
These times will come from sequence events, when present (which will be the usual case).

Reply:
Same here.